### PR TITLE
[KOGITO-1839] - Allow user use Make command to build and push images …

### PR DESCRIPTION
**Pull request by spolti**
…to local registry

Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

File: scripts/push-local-registry.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace=${3:-openshift}
+
+if [ "${registry}x" == "x"  ]; then
+    echo "No registry provided, please set the env REGISTRY or set it as parameter to this script"
+    echo "Simple usage: /bin/sh scripts/push-local-registry.sh ${REGISTRY} ${SHORTENED_LATEST_VERSION} ${NS}"
+    exit 1
+fi
+if [ "${version}" == "latest"  ]; then
+    echo "No version provided, latest will be used"
+    exit 1
```
Comment (sutaakar) [inline scripts/push-local-registry.sh]: I think that the exit shouldn't be here

File: scripts/push-local-registry.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+fi
+if [ "${namespace}x" == "x"  ]; then
+    echo "No registry provided, images will be installed on openshift namespace"
+    exit 1
```
Comment (sutaakar) [inline scripts/push-local-registry.sh]: same here

File: scripts/push-local-registry.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    echo "No registry provided, please set the env REGISTRY or set it as parameter to this script"
+    echo "Simple usage: /bin/sh scripts/push-local-registry.sh ${REGISTRY} ${SHORTENED_LATEST_VERSION} ${NS}"
+    exit 1
+fi
+if [ "${version}" == "latest"  ]; then
+    echo "No version provided, latest will be used"
+    exit 1
+fi
+if [ "${namespace}x" == "x"  ]; then
+    echo "No registry provided, images will be installed on openshift namespace"
```
Comment (sutaakar) [inline scripts/push-local-registry.sh]:
```suggestion
    echo "No namespace provided, images will be installed on openshift namespace"
```

File: scripts/push-local-registry.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+if [ "${registry}x" == "x"  ]; then
+    echo "No registry provided, please set the env REGISTRY or set it as parameter to this script"
+    echo "Simple usage: /bin/sh scripts/push-local-registry.sh ${REGISTRY} ${SHORTENED_LATEST_VERSION} ${NS}"
+    exit 1
+fi
+if [ "${version}" == "latest"  ]; then
+    echo "No version provided, latest will be used"
+    exit 1
+fi
+if [ "${namespace}x" == "x"  ]; then
```
Comment (sutaakar) [inline scripts/push-local-registry.sh]:
```suggestion
if [ "${namespace}" == "openshift"  ]; then
```

File: scripts/push-local-registry.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    echo "No registry provided, images will be installed on openshift namespace"
+    exit 1
+fi
+
+echo "Images version ${version} will be pushed to registry ${registry}"
+
+for image in ${IMAGES}; do
+    echo "tagging image ${image} to ${registry}/${namespace}/${image}:${version}"
+    ${BUILD_ENGINE} tag quay.io/kiegroup/${image}:${version} ${registry}/${namespace}/${image}:${version}
+    echo "Deleting imagestream ${image} if exists `oc delete oc -n ${namespace} ${image}`"
```
Comment (sutaakar) [inline scripts/push-local-registry.sh:30]: This log doesn't correspond with the step below.

File: scripts/push-local-registry.sh (hunk 1 above)
Comment (spolti) [reply scripts/push-local-registry.sh]: +1

File: scripts/push-local-registry.sh (hunk 2 above)
Comment (spolti) [reply scripts/push-local-registry.sh]: +1

File: scripts/push-local-registry.sh (hunk 3 above)
Comment (spolti) [reply scripts/push-local-registry.sh]: ctrl c + v leftover :D

File: scripts/push-local-registry.sh (hunk 5 above)
Comment (spolti) [reply scripts/push-local-registry.sh:30]: actually the step is in the same line, note the command after the word exists.

**Review by ricardozanini — APPROVED** [review]

File: Makefile [hunk 6]
```diff
@@ -108,3 +108,13 @@ push-staging: build _push-staging
 _push-staging:
 	python3 scripts/push-staging.py
 
+# push to local registry, useful to push the built images to local registry
+# requires parameter: REGISTRY: my-custom-registry:XXXX
```
Comment (radtriste) [inline Makefile]: what does the `XXXX` stands for ?

**Review by radtriste — APPROVED** [review]
small comment, otherwise ok for me

File: Makefile (hunk 6 above)
Comment (spolti) [reply Makefile]: the registry port, if needed. let me improve it.

**Review by radtriste — APPROVED** [review]

File: scripts/push-local-registry.sh (hunk 5 above)
Comment (sutaakar) [reply scripts/push-local-registry.sh:30]: oh, ok, I overlooked it.

**Review by sutaakar — APPROVED** [review]

**Pull request merged**